tests/phpunit/CRMTraits/Custom/CustomDataTrait.php

```diff
@@ -42,7 +49,7 @@ public function createCustomGroup($params = []) {
       'max_multiple' => 0,
     ], $params);
     $identifier = $params['name'] ?? $params['title'];
-    $this->ids['CustomGroup'][$identifier] = $this->callAPISuccess('CustomGroup', 'create', $params)['id'];
+    $this->ids['CustomGroup'][$identifier] = CustomGroup::create()->setValues($params)->execute()->first()['id'];
```

Comment:
> (From MatterMost) I feel like we should be using apiv4 where we can now for everything other than testing apiv3 - but our reliance / comfort with helper functions makes that surprisingly hard

For this kind of thing - setting up a couple records - I agree that it makes more sense to start using APIv4 instead of APIv3. +1

IMHO, it's nice that `civicrm_api4()` doesn't seem to wrap-up exceptions in the same way that `civicrm_api3()` does. So I don't think we need an analog for `callAPISuccess()`?

Regarding test-helpers generally... it's not monolithic IMHO. There probably are some that could be replaced with API calls. But consider a helper like `individualCreate()` or `participantCreate() which bakes-in some defaults/examples. These are not meaningful to any general audience, so it doesn't feel right to present them as a public "API".  Never-the-less, they're useful in writing tests (because it otherwise gets quite redundant to define examples).

Reply:
@totten I was imagining 'fixing' individualCreate to itself call apiv4 though

Reply:
Yup, that makes sense. :+1: